post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java

```diff
@@ -312,13 +317,14 @@ private String extractValFromMessage(String topic, String payload) {
                 }
             } else if (topic.endsWith(Entity.OBSERVATION.getName())) {
                 String domainCd = objectMapper.readTree(payload).get(PAYLOAD).path("obs_domain_cd_st_1").asText();
-                String ctrlCd = objectMapper.readTree(payload).get(PAYLOAD).path("ctrl_cd_display_form").asText();
+                String ctrlCd = Optional.ofNullable(objectMapper.readTree(payload).get("ctrl_cd_display_form"))
+                        .filter(node -> !node.isNull()).map(JsonNode::asText).orElse(null);
```

Comment:
Suggestion:  you can consider using `ObjectUtils.isEmpty` which has enhanced null pointer chekcks.

Reply:
> Suggestion: you can consider using `ObjectUtils.isEmpty` which has enhanced null pointer chekcks.

Unfortunately in this case the node is not null, it's NullNode class.